### PR TITLE
Model documentation

### PR DESCRIPTION
**Pull request by Jegp**
* Retrofitted the documentation to same structure as main page\
* Added descriptions on how to get started with Norse
* Minor fixes and typos

### CODE REVIEW

File: docs/source/hardware.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _page-hardware:
 
 Hardware acceleration
--------------------
+---------------------
```
Comment (cpehle) [inline docs/source/hardware.rst:4]: nice this has been bothering me for a while....

File: docs/source/index.rst [hunk 2]
```diff
@@ -1,8 +1,10 @@
+.. page-index:
```
Comment (cpehle) [inline docs/source/index.rst:1]: Do you know what these GitHub Actions mean?

File: docs/source/learning.rst [hunk 3]
```diff
@@ -3,9 +3,81 @@
...
+
+Training Spiking Neural Networks
+=====================================
+
+Many solutions have been attempted to solve this problem with varying 
+success. 
+Here we will only cover the *surrogate gradient* approach, and
+illustrate how  it is implement in norse.
+
+Our approach builds on the SuperSpike method proposed by
```
Comment (cpehle) [inline docs/source/learning.rst:31]: This is not entirely true. To the best of my knowledge it was first introduced by https://www.pnas.org/content/113/41/11441

Comment (cpehle) [reply docs/source/learning.rst:31]: There is a review by Neftci, Zenke and Mostafa of "Surrogate Gradient methods"

**Review by cpehle — APPROVED** [review]
Really nice work overall, we should be a bit more careful with attributing the methods, maybe we should start a collection of references.

File: docs/source/index.rst (hunk 2 above)
Comment (Jegp) [reply docs/source/index.rst:1]: I would think GitHub is clever enough to link warnings/errors to the code. This is probably one... Of many. I only cleaned up ~30% of the errors. There are still so many left. Especially from absail.

Comment (cpehle) [reply docs/source/index.rst:1]: Yeah abseil, wasn't a great idea....

Comment (Jegp) [reply docs/source/index.rst:1]: Well. It's a start. Did you see the argparse flags that Pytorch Lightning introduces? I really like that approach

**Pull request merged**